platforms/cpuinfo.go

```diff
@@ -0,0 +1,56 @@
+package platforms
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+)
+
+// For Linux, the kernel has already detected the ABI, ISA and Features.
+// So we don't need to access the ARM registers to detect platform information
+// by ourselves. We can just parse these information from /proc/cpuinfo
+func getCpuInfo(pattern string) (info string, err error) {
+	if !isLinuxOS(runtime.GOOS) {
+		return "", fmt.Errorf("Unsupported Operation System: %s", runtime.GOOS)
+	}
+
+	cpuinfo, err := os.Open("/proc/cpuinfo")
+	if err != nil {
+		return "", err
+	}
+
+	defer cpuinfo.Close()
+
+	// Start to Parse the Cpuinfo line by line. For SMP SoC, we parse
+	// the first core is enough.
+	scanner := bufio.NewScanner(cpuinfo)
+	for scanner.Scan() {
+		newline := scanner.Text()
+		list := strings.Split(newline, ":")
+
+		if len(list) > 1 && strings.EqualFold(strings.TrimSpace(list[0]), pattern) {
+			return strings.TrimSpace(list[1]), nil
+		}
+	}
+
+	// Check whether the scanner encountered errors
+	err = scanner.Err()
+	if err != nil {
+		return "", err
+	}
+
+	return "", fmt.Errorf("Could not find value for pattern: %s", pattern)
+}
+
+func getCPUVariant() string {
+	variant, err := getCpuInfo("Cpu architecture")
+	if err == nil {
+		return "v" + variant
+	}
+	return "unknown"
+}
+
+// Present the ARM instruction set architecture, eg: v7, v8
+var armCpuVariant = getCPUVariant()
```

Comment:
My only concern here is we are parsing `/proc/cpuinfo` on any Linux OS as the code is right now. I would rather this be a "lazy" instantiation so it never runs unless needed.

Reply:
Yes, this function call is useless for other architectures/OS. I would address this comment in next update.